AnkiHubSoftware/ankihub_addon
# fix: Check for note type field mismatch, disallow suggestions with mismatching fields

## Related issues

Discussion: https://community.ankihub.net/t/why-did-i-get-this-error/5941/8


## Proposed changes
When creating suggestions, check the note type fields in the Anki database against the note type fields in the AnkiHub database. 
If the fields differ, the add-on shows a dialog that tells user that the fields were changed, tells them changing them is currently not supported, and ask them if they want to reset the note type.

<img src="https://github.com/ankipalace/ankihub_addon/assets/31575114/dbc9219a-c439-445c-9825-cc1e4b9b1d83" width="500" />

## Todo
- in another PR: add performance test for exporting notes to see if  the field mismatch check slows things down
  - if performance is worse, consider creating a function that has the functionality of `ankihub.main.exporting.to_note_data` but takes a list of Anki notes and checks for note type mismatches for all passed notes first
- when the note type is reset, this doesn't affect the templates yet, only the fields; user could potentially end up with broken templates
- refresh the editor after resetting the note type so that it displays the updated fields


## How to reproduce
Explain like if someone who doesn't know this project is reviewing your changes and how they can replicate this.

- Add a new field to an AnkiHub note type
- Edit an AnkiHub note of that note type
- Suggest a change to the note
- The dialog shown above should open
- Choose Yes to reset the note type
